src/app/main/component/ubs/components/ubs-order-details/ubs-order-details.component.ts

```diff
@@ -245,16 +248,25 @@ export class UBSOrderDetailsComponent implements OnInit, OnDestroy {
         this.orderService
           .processCertificate(certificate)
           .pipe(takeUntil(this.destroy))
```

Comment:
calculateCertificates -> this sends multiple requests 
for old certificates and new ones 
probably because of for (const certificate of arr) { ..
looks like you need clean certificatesArray somwhere

Reply:
![image](https://user-images.githubusercontent.com/43093994/120118389-f71e1400-c19a-11eb-9c5a-dbc5af755ca2.png)